test/src/unit-cppapi-config.cc

```diff
@@ -130,9 +130,52 @@ TEST_CASE(
   CHECK(result3 == value3);
 }
 
+TEST_CASE(
+    "C++ API: Config Profile Selection via Env and Config",
+    "[cppapi][config][profile]") {
+  // Create two profiles with different values for the same key.
+  const std::string key = "rest.server_address";
+  const std::string profile1_name = "profile1";
+  const std::string profile2_name = "profile2";
+  const std::string profile1_value = "profile1_localhost:8080";
+  const std::string profile2_value = "profile2_localhost:8080";
+
+  tiledb::sm::TemporaryLocalDirectory tempdir_;
+  const std::string profile_dir = tempdir_.path();
+
+  // Create profile1
+  auto profile1 = tiledb::Profile(profile1_name, profile_dir);
+  profile1.set_param(key, profile1_value);
+  profile1.save();
+
+  // Create profile2
+  auto profile2 = tiledb::Profile(profile2_name, profile_dir);
+  profile2.set_param(key, profile2_value);
+  profile2.save();
+
+  // Set the environment variable to select profile1
+  setenv_local("TILEDB_PROFILE_NAME", profile1_name.c_str());
```

Comment:
@kounelisagis was there going to be env var for the profiles location/directory too?